### ISSUE TITLE
undefined variables

### ISSUE THREAD

**Issue by sunilw** [simanek/Heads-Up-Grid#9]
When loading a page with hug enabled, console gives the error "gridonload is not defined".  Javascript execution then stops.

When toggling the gird via the button, I get "gridstate is not defined", and script execution stops.

This is on firefox 13.01.

These errors can be seen on the heads up grid homepage.

**Comment by simanek** [comment]
I'm not seeing this problem on Firefox 13.0.1 or Chromium right now. Grid displays and turns on and off just fine. Curious. Plus I haven't made any changes in the last 14 days or so. When did you begin encountering this problem?

**Comment by sunilw** [comment]
Curious indeed.

More details... 
I'm assuming that the application homepage is using the latest version, so I'll go into detail about what I see there. The issues happen to be the same issues that I get when I use hugrid locally.

First I clear my cache, clear the firebug error, and then shift-reload. I then see this in the console:

'gridonload is not defined'
'if (gridonload === 'off') {'
'hugrid.js (line 124)'

After this point, other scripts attached to the  page fail to execute.

The other error happens under two circumstances:
If I have the hugrid page open in a tab, and select another tab. When I return to the hugrid homepage tab, the new error shortly appears:

'gridstate is not defined'
'if (gridstate === 'off') {'
'hugrid.js (line 134)'

The other circumstance: This error also appears if I toggle the grid. fb gives a different line number though:
hugrid.js (line 134)

I don't use chrome much, so I don't know much about debugging using it's tools. I'm using version "18.0.1025.151 (Developer Build 130497 Linux) Ubuntu 12.04"

Under this version, if open the console, I get similair errors.
Upon opening the page, I see this:

Uncaught ReferenceError: gridonload is not defined  hugrid.js: 124

hugrid.js:134  Uncaught ReferenceError: gridstate is not defined

And, when toggling the grid:
Uncaught ReferenceError: gridstate is not defined hugrid.js:114

So, these look like the same errors.

Let me know if I can provide any more information, or do anymore testing on my end of things.

**Comment by simanek** [comment]
Okay. I thought you were saying that the Heads-Up Grid wasn't working
and that you were seeing these errors. But after looking in the
browser console inspector while the page loads, I see the errors you
are referring to.

The plugin is working correctly, but obviously my JavaScript is doing
something wrong. No doubt this is due to my massive ignorance about
programming and scripting. :D

I'll do some reading. Maybe this is due to how I am defining
variables? Maybe I need to include more conditional statements to
account for these undefined variable errors?

On Thu, Jul 5, 2012 at 5:52 PM, sunil
reply@reply.github.com
wrote:

> First I clear my cache, clear the firebug error, and then shift-reload. I then see this in the console:
> 
> 'gridonload is not defined'
> 'if (gridonload === 'off') {'
> 'hugrid.js (line 124)'

**Comment by sunilw** [comment]
Well, I can try to give you a hand, but I'm better at finding bugs then fixing them.

If we look the variable definitions for both setgridonload and setgridonresize, they both have 'if' statements that check variables that their respective functions can't find.

line 124: 
if (gridonload === 'off') {...}

but I can't see gridonload anywhere else in the code.

Maybe I should have split this into a second issue... but with the error "gridstate is not defined", I think I can see what's going on there.

line 133 defines the variable setgridonresize. It begins by checking the value of 'gridstate'. But it can't find it, so it errors out and execution stops.

I think the problem here is one of scope. 

I find 'gridstate' as a private member variable of an anonymous function earlier in the script. 
"$('#hugridButton').click(function () {...}"
Because it's in the scope of that function, other functions can't access it. 

I'm guessing that 'gridstate' should be on of off depending on whether we can see the grid. Is this right?

If I'm right, then both of these variables should be defined and accessible and settable from within the plugin. 

I'm kinda new at helping other people debug their code, so let me know if this helpful.

**Comment by simanek** [comment]
Thanks for the help. Yeah... what's weird is that the variables as I have created them WORK as expected. Both gridonload and gridstate work correctly (gridonload defines whether the grid is initially on or off on pageload, gridstate preserves the state of the grid as the browser window is resized.)

So I need to do some learning. The JS as I wrote it works very well, but these errors you've pointed out are obviously not ideal. They may affect this script's efficiency or create conflicts with other scripts on the page potentially.

Fortunately this is the first I have heard of these problems, so hopefully there are no problems with the actual functionality of the script. There's probably some really important lesson to be learned in fixing these errors!
